test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -38,11 +57,12 @@ struct test_inclusive_scan_by_segment
         //T keys[n1] = { 1, 2, 3, 4, 1, 1, 2, 2, 3, 3, 4, 4, 1, 1, 1, ...};
         //T vals[n1] = { 1, 1, 1, ... };
 
-        int segment_length = 1;
-        int i = 0;
-        while (i != n)
+        using DifferenceType = typename DifferenceTypeT<decltype(host_vals)>::DifferenceType;
+
```

Comment:
I would remove the struct `DifferenceTypeT `and use just one line here:
`using DifferenceType = typename ::std::decay<decltype(host_vals[0])>::type;` 
Or even better - just deduced Size type, instead of DifferenceType.

Reply:
Fixed